helix-core/src/main/java/org/apache/helix/HelixManagerProperties.java

```diff
@@ -116,6 +116,8 @@ static boolean versionNoLessThan(String version1, String version2) {
 
         if (versionNum1 < versionNum2) {
           return false;
+        } else if (versionNum1 > versionNum2) {
```

Comment:
This just ignore the check, right? Can we make the logic correct here, say 1.0.x > 0.9.x?

Reply:
This is not to ignore the check. This is a digit by digit comparison. See above code, the version is already split. So we are comparing "1" to "0", which breaks the loop. If it's "0" to "0", we will continue to compare next digit. It should be complete.

Reply:
Previous error is just that it didn't break even the digit is larger the minimum version. I guess it didn't consider we might bump to 1.0.0.0.